commit 44674d9c2267f454f38df7b2395939bfa911f92e
Author: Ayala Beker <ayala.beker@intel.com>
Date:   Wed Sep 23 10:41:27 2015 +0200

    mac80211: advertise support for full station state in AP mode
    
    This enables adding stations in unauthenticated mode, just after
    receiving the first authentication frame; which in turn allows
    sending a negative authentication reply if the station cannot be
    added.
    
    In addition init rate control for unassociated station only when
    it becomes associated, prior to that low rates will be used.
    
    Signed-off-by: Ayala Beker <ayala.beker@intel.com>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/main.c b/net/mac80211/main.c
index ff79a13d231d..9b813a2f3a75 100644
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -543,7 +543,8 @@ struct ieee80211_hw *ieee80211_alloc_hw_nm(size_t priv_data_len,
 			   NL80211_FEATURE_HT_IBSS |
 			   NL80211_FEATURE_VIF_TXPOWER |
 			   NL80211_FEATURE_MAC_ON_CREATE |
-			   NL80211_FEATURE_USERSPACE_MPM;
+			   NL80211_FEATURE_USERSPACE_MPM |
+			   NL80211_FEATURE_FULL_AP_CLIENT_STATE;
 
 	if (!ops->hw_scan)
 		wiphy->features |= NL80211_FEATURE_LOW_PRIORITY_SCAN |